### ISSUE TITLE
Allow SQLDriverConnect instead of SQLConnect

### ISSUE THREAD

**Issue by psykora** [Erroneous1/sqlpp11-connector-odbc#9]
Currently the ODBC connector is using `SQLConnect` which requires datasource to be defined in the system (Windows) or ini file (Unix).

It would be nice to be able to specify connection string during runtime. Something like this (example for MSSQL server)
`Driver={SQL Server};Server=SERVER_HOSTNAME\INSTANCE;Database=DBNAME;Uid=DBUSER;Pwd=DBPASS;`

**Comment by Erroneous1** [comment]
This looks like a great idea. Looking at the code, I think detail::connection_handle_t doesn't need any of the information that's only used for SQLConnect so there is a bit of work to be done there and then just add some alternate config structs and connection constructors. I might work on it this week.

I'll be adding constructors for `SQLDriverConnect` and `SQLBrowseConnect`.

**Comment by Erroneous1** [comment]
@psykora I made a branch feature/alternate_connects which adds `driver_connection_config`. Can you test it out to see if it works for you?

There are some breaking changes. ODBC_Type is now in sqlpp11::odbc instead of `sqlpp11::odbc::connection_config`. There isn't a `connection_config.database` anymore (I could add back in but it seems out of place, open for discussion). I need to work on CMake to get it to work better with sqlpp11 and date, maybe include both as subprojects.

**Comment by psykora** [comment]
@Erroneous1: Thanks! I didn't expect so prompt action. 👍 
I am just experimenting with sqlpp11 and wanted to test it with _MSSQL_. I noticed your odbc connector and hardly managed to build _master_ with visual studio 2017 15.6 and now my _cmake_ parameters are not working again :). Any reason why is it is not based on _master_?

**Comment by psykora** [comment]
@Erroneous1 I cherry-picked your commit into master and the driver connection works great!

**Comment by psykora** [comment]
Btw. here is what I had to change to pass compilation under MSVC 2017 15.6
https://github.com/psykora/sqlpp11-connector-odbc/commit/14d2aa7c5e818c2eee54c0aa8e7226f3219f1b13

my cmake configure command:
`cmake .. -DCMAKE_PREFIX_PATH=C:\work\inst -DHinnantDate_ROOT_DIR=C:\work\inst\date_prj -DCMAKE_INSTALL_PREFIX=C:\work\inst\sqlpp11-connector-odbc`

**Comment by Erroneous1** [comment]
Yeah I accidentally branched from an old branch instead of master. I'll be
making a new branch soon with the same interface but from master. That's
why I was having cmake issues as well.

**Comment by Erroneous1** [comment]
Ok try out feature/SQLDriverConnect instead. This is modified from master instead of an old branch. I am also making giving it a string reference optional since you don't need it if you know the config is correct anyway.

**Comment by psykora** [comment]
It works well, I will send you several pull requests and you can check whether it still compiles with clang. I am compiling with MSVC and some of the changes are necessary to pass compilation.
Thanks!
